addon/components/course-materials.js

```diff
@@ -19,17 +21,20 @@ export default class CourseMaterialsComponent extends Component {
   }
 
   @restartableTask
-  *load(element, [
-    courseLearningMaterials,
-    sessions
-  ]) {
-    if (courseLearningMaterials) {
-      this.courseLearningMaterialObjects = yield map(courseLearningMaterials.toArray(), async (clm) => {
+  *load() {
```

Comment:
This is only the first step in an eventual conversion to [use](https://github.com/pzuraq/ember-could-get-used-to-this) and resources which would remove the `load()` method (and render modifiers) in favor of a more `get()` type syntax. If I'm understanding it correctly we would have something like 

```
@use courseMaterials = new ResolveRelationship(() => [
  this.args.course.learningMaterials
]);

get materialTitles(){
  return this.materials.mapBy('title');
}
```

Then the template would not actually be involved at all.  So while I do like the more explicit invocation it will add an extra step when removing these in favor of the hopefully idiomatic future ember pattern so I think this is better to merge as is and then move forward. It's not a big change either way though so I'm happy to add this if you feel like it ads good interim value.

Reply:
fair enough, thanks for this preview of things to come.